packages/dds/tree/src/test/util/arrayUtilities.spec.ts

```diff
@@ -0,0 +1,170 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import { validateUsageError } from "@fluidframework/test-runtime-utils/internal";
+
+import {
+	validateIndex,
+	validateIndexRange,
+	validatePositiveIndex,
+	validateSafeInteger,
+} from "../../util/index.js";
+
+describe("arrayUtilities unit tests", () => {
+	it("validateSafeInteger", () => {
+		assert.doesNotThrow(() => validateSafeInteger(0, "test"));
+		assert.doesNotThrow(() => validateSafeInteger(-0, "test"));
+		assert.doesNotThrow(() => validateSafeInteger(1, "test"));
+		assert.doesNotThrow(() => validateSafeInteger(-1, "test"));
+		assert.doesNotThrow(() => validateSafeInteger(Number.MAX_SAFE_INTEGER, "test"));
+		assert.doesNotThrow(() => validateSafeInteger(Number.MIN_SAFE_INTEGER, "test"));
+
+		assert.throws(
+			() => validateSafeInteger(1.5, "test"),
+			validateUsageError(/Expected a safe integer passed to test, got 1.5/),
+		);
+		assert.throws(
+			() => validateSafeInteger(Number.MAX_SAFE_INTEGER + 1, "test"),
+			validateUsageError(/Expected a safe integer passed to test, got 9007199254740992/),
+		);
+		assert.throws(
+			() => validateSafeInteger(Number.MIN_SAFE_INTEGER - 1, "test"),
+			validateUsageError(/Expected a safe integer passed to test, got -9007199254740992/),
+		);
```

Comment:
It's fine, but interesting note: adding 1 to MAX_SAFE_INTEGER is kind of sketchy. You get the same value if you add 2 to it for example. I'm a bit surprised JS didn't set MAX_SAFE_INTEGER one higher at 9007199254740992 in which case this test would fail.

As the failure mode here would be the test failing and giving a false error, I think this is fine, but I'm surprised it worked.